recipes/aravis/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+import shutil
+
+
+class AravisConan(ConanFile):
+    name = "aravis"
+    license = "LGPL-2.1-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/AravisProject/aravis"
+    description = "A vision library for genicam based cameras."
+    topics = ("usb", "camera")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "usb": [True, False],
+        "packet_socket": [True, False],
+        "gst_plugin": [True, False],
+        "tools": [True, False],
+        "introspection": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "usb": True,
+        "packet_socket": True,
+        "gst_plugin": False,
+        "tools": True,
+        "introspection": False
+    }
+    generators = "pkg_config"
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _aravis_api_version(self):
+        return ".".join(self.version.split(".")[0:2])
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.packet_socket
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```

Comment:
```suggestion
            del self.options.fPIC
           self.options["glib"].shared = True
```

I would enforce it, otherwise, CCI won't generate packages on Linux.

Reply:
Makes sense, fixed 👍